plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchTaskHandler.groovy

```diff
@@ -115,13 +116,17 @@ class AzBatchTaskHandler extends TaskHandler implements FusionAwareTask {
         final done = taskState0(taskKey)==BatchTaskState.COMPLETED
         if( done ) {
             // finalize the task
-            task.exitStatus = readExitFile()
+            final info = batchService.getTask(taskKey).executionInfo
+            task.exitStatus = info?.exitCode ?: readExitFile()
             task.stdout = outputFile
             task.stderr = errorFile
             status = TaskStatus.COMPLETED
-            final info = batchService.getTask(taskKey).executionInfo
-            if (info.result == BatchTaskExecutionResult.FAILURE)
-                task.error = new ProcessUnrecoverableException(info.failureInfo.message)
+            if (info.result == BatchTaskExecutionResult.FAILURE ) {
+                if (task.exitStatus == 0 || task.exitStatus == Integer.MAX_VALUE) {
```

Comment:
The MAX_VALUE condition is managed here 

https://github.com/nextflow-io/nextflow/blob/076c8237afa51b219d140ccdcba121fd6e5f0e28/modules/nextflow/src/main/groovy/nextflow/processor/TaskProcessor.groovy#L2385-L2385

Reply:
I didn't see it. Updated

Reply:
I have reopened it because I have noted that if we do not manage the MAX_VALUE case here we will loose the error message provided by Azure. @pditommaso, if you agree I will revert the change